### PR TITLE
feat: avoid Instantiation of banker on pure packages

### PR DESCRIPTION
**Pull request by Villaquiranm**
Related to #2192 
- Trigger a panic if a p/ attempts to create a banker, but allow interaction with an already instantiated banker from r/.

Don't really know if this is the right approach. If you have another idea of how to implement this security check please let me know :)

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Review by thehowl — CHANGES_REQUESTED** [review]
This will not work in a real life scenario, as `CurrentRealm()` skips frames of pure packages. (If you want a good side-task, change `NewCodeRealm` to work only with PkgPath's that match the rules of `gnolang.IsRealmPath`)

I suggest:

- Having a function like `assertCallerIsRealm`.
- This should get the frame of the function where this is called (I think m.Frames[m.NumFrames()-2]).
- Then call `LastPackage` and make sure its PkgPath matches that of gnolang.IsRealmPath.

Thanks!

**Comment by Villaquiranm** [comment]
> This will not work in a real life scenario, as `CurrentRealm()` skips frames of pure packages. (If you want a good side-task, change `NewCodeRealm` to work only with PkgPath's that match the rules of `gnolang.IsRealmPath`)
> 
> I suggest:
> 
> * Having a function like `assertCallerIsRealm`.
> * This should get the frame of the function where this is called (I think m.Frames[m.NumFrames()-2]).
> * Then call `LastPackage` and make sure its PkgPath matches that of gnolang.IsRealmPath.
> 
> Thanks!

Hello thanks a lot for your review and for the great guidance :)
I hope I got the general idea of what you wanted me to do, I got a few questions though
-  I put this code on **assertCallerIsRealm** function should I make some extra verifications ? I don't really know why m.NumFrames()-2.
```go
frame := m.Frames[m.NumFrames()-2]
if !frame.LastPackage.IsRealm() {
	m.Panic(typedString("caller is not a realm"))
}
```
- When you said a side task for `NewCodeRealm` this is what you meant ? It is okay to add another function on sdt tests ?
[e7ade2c](https://github.com/gnolang/gno/pull/2248/commits/e7ade2cd4e895b638de4d43bd5940d8ff7217ea0)

**Review by thehowl — APPROVED** [review]

File: gnovm/tests/stdlibs/std/frame_testing.gno [hunk 1]
```diff
@@ -8,5 +8,8 @@ func NewUserRealm(user Address) Realm {
 // NewCodeRealm creates a new realm for a published code realm with the given
 // pkgPath.
 func NewCodeRealm(pkgPath string) Realm {
+	if !isRealm(pkgPath){
```
Comment (thehowl) [inline gnovm/tests/stdlibs/std/frame_testing.gno:11]: fmt

Comment (Villaquiranm) [reply gnovm/tests/stdlibs/std/frame_testing.gno:11]: I did a make fmt but I did not change the files

**Review by moul — APPROVED** [review]

File: gnovm/tests/stdlibs/std/frame_testing.gno (hunk 1 above)
Comment (thehowl) [reply gnovm/tests/stdlibs/std/frame_testing.gno:11]: Can you update `gnovm/Makefile`'s `fmt` rule with the following?

```
	go run ./cmd/gno fmt $(GNOFMT_FLAGS) ./stdlibs/... ./tests/files/extern/... ./tests/stdlibs/...
```

This directory was skipped.

Then run `make fmt` and we should be good to go.

Comment (Villaquiranm) [reply gnovm/tests/stdlibs/std/frame_testing.gno]: fixed on bb91f0175e328fbc3fe6c6dcdaafcfbb6efab5f3

**Comment by thehowl** [comment]
Apologies, revert the changes in tests/files/extern and remove that dir from the fmt rule.

`gno fmt` doesn't understand the imports in that dir and it shouldn't, so let's just skip it.

**Comment by Villaquiranm** [comment]
> Apologies, revert the changes in tests/files/extern and remove that dir from the fmt rule.
> 
> `gno fmt` doesn't understand the imports in that dir and it shouldn't, so let's just skip it.

No worries :)
fixed on 53369df58bfec1d94b0f2d350f0ccd6668982336

**Pull request merged**